core/metadata.jvm/src/org/jetbrains/kotlin/metadata/jvm/deserialization/ClassMapperLite.kt

```diff
@@ -63,7 +63,7 @@ object ClassMapperLite {
             add("reflect/KFunction$i", "kotlin/reflect/KFunction")
         }
 
-        for (klass in listOf("Char", "Byte", "Short", "Int", "Float", "Long", "Double", "String", "Enum")) {
+        for (klass in listOf("Char", "Byte", "Short", "Int", "Float", "Long", "Double", "String", "Enum", "Boolean")) {
```

Comment:
This change will probably confuse the compiler and reflection of older versions when they encounter a declaration with `kotlin.Boolean.Companion` in the signature. The generated bytecode will be correct, but annotations (such as `Deprecated`, for example) will not be loaded, and reflective call will throw an exception. For more information, see the comment on this class.

Please revert this change and add a short comment noting that we explicitly omit `Boolean` here. Thanks!

Reply:
Done